builder/azure/arm/config.go

```diff
@@ -65,7 +64,7 @@ var (
 	reResourceGroupName    = regexp.MustCompile(validResourceGroupNameRe)
 	reSnapshotName         = regexp.MustCompile(`^[A-Za-z0-9_]{1,79}$`)
 	reSnapshotPrefix       = regexp.MustCompile(`^[A-Za-z0-9_]{1,59}$`)
-	reResourceNamePrefix   = regexp.MustCompile(validResourceNamePrefix)
+	reResourceNamePrefix   = regexp.MustCompile(`^[A-Za-z0-9][A-Za-z0-9-]{1,9}$`)
```

Comment:
Is there any reason why we reject names starting with `-`? From what I see in the docs it seems to be supported, unless I'm wrong, if so do correct me.
If that's so though, we can change the regex to `[A-Za-z0-9-]{1,10}`

Reply:
Come to think of it, if we want a one-character prefix, we can't accept it with your regex since it introduces a minimum of 2 characters, maybe the second part should be `{0,9}` if we maintain the logic behind validation

Reply:
> Is there any reason why we reject names starting with `-`? From what I see in the docs it seems to be supported, unless I'm wrong, if so do correct me. If that's so though, we can change the regex to `[A-Za-z0-9-]{1,10}`

I couldn't find anything in the documentation either, but when I try to create a VM starting with `-`, I get an error. So it's probably a parent rule that decides that: 
![image](https://user-images.githubusercontent.com/32768682/181843224-5293eee7-f0e1-478d-8e9a-631ce828cbd3.png)